pyproject.toml

```diff
@@ -15,6 +15,13 @@ astor = "^0.8.1"
 openai = "^0.27.5"
 ipython = "^8.13.1"
 matplotlib = "^3.7.1"
+google-auth = "^2.18.0"
```

Comment:
we are forcing to install Google library even if one does not use Google Sheets, this is a problem for the future of the **pandasai** project, we will have N LLM and multiple output and input support (from various projects), this is not healthy for **pandasai**

we can use [extra pip dependency](https://peps.python.org/pep-0508/#extras), [poetry documentation](https://python-poetry.org/docs/pyproject/#extras)

cc @gventuri

Reply:
@avelino, I agree, we should find a compromise to keep it lightweight!

@aryandgandhi what do you think about adding it in `[tool.poetry.extras]` instead? Check out [poetry documentation](https://python-poetry.org/docs/pyproject/#extras)